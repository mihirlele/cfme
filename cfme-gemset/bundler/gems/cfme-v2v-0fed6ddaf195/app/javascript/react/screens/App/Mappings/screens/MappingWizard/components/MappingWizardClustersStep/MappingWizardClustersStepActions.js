import URI from 'urijs';
import API from '../../../../../../../../common/API';
import { FETCH_V2V_SOURCE_CLUSTERS, QUERY_V2V_PROVIDERS } from './MappingWizardClustersStepConstants';

export { showAlertAction, hideAlertAction } from '../../MappingWizardActions';

const _getSourceClustersActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_SOURCE_CLUSTERS,
    payload: API.get(url)
  });

export const fetchSourceClustersAction = url => {
  const uri = new URI(url);
  return _getSourceClustersActionCreator(uri.toString());
};

// ****************************************************************************
// QUERY_V2V_PROVIDERS
// ****************************************************************************
const _queryProvidersActionCreator = (url, providerIds) => dispatch =>
  dispatch({
    type: QUERY_V2V_PROVIDERS,
    payload: API.post(url, {
      action: 'query',
      resources: providerIds.map(id => ({ id }))
    })
  });

export const queryProvidersAction = (url, providerIds) => {
  const uri = new URI(url);
  uri.addSearch({ attributes: 'authentications' });

  return _queryProvidersActionCreator(uri.toString(), providerIds);
};
