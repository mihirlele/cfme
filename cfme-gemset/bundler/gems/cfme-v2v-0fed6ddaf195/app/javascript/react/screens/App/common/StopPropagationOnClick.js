import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'patternfly-react';

// This component has no role on the page, and will be hidden from screen readers (role="presentation").
// Prevents a click on an inner element (e.g. a DropdownKebab) from triggering a click on an ancestor (e.g. a table row).
// The onKeyUp handler and role are used to satisfy eslint-plugin-jsx-a11y.
const StopPropagationOnClick = ({ children }) => (
  <span onClick={e => e.stopPropagation()} onKeyUp={noop} role="presentation">
    {children}
  </span>
);

StopPropagationOnClick.propTypes = {
  children: PropTypes.node
};

export default StopPropagationOnClick;
