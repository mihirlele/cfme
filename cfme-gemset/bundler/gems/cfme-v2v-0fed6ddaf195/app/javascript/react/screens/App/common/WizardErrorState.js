import React from 'react';
import PropTypes from 'prop-types';

const WizardErrorState = ({ title, message, onClose }) => (
  <div className="wizard-pf-complete blank-slate-pf">
    <div className="wizard-pf-success-icon">
      <span className="pficon pficon-error-circle-o" />
    </div>
    <h3 className="blank-slate-pf-main-action">{title}</h3>
    <p className="blank-slate-pf-secondary-action">{message}</p>
    {onClose ? (
      <button type="button" className="btn btn-lg btn-primary" onClick={onClose}>
        {__('Close')}
      </button>
    ) : null}
  </div>
);

WizardErrorState.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string,
  onClose: PropTypes.func
};

WizardErrorState.defaultProps = {
  message: ''
};

export default WizardErrorState;
