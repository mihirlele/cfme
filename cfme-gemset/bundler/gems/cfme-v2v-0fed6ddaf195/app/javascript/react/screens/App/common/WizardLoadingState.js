import React from 'react';
import PropTypes from 'prop-types';
import { Spinner } from 'patternfly-react';

const WizardLoadingState = ({ title, message }) => (
  <div className="wizard-pf-process blank-slate-pf">
    <Spinner loading size="lg" className="blank-slate-pf-icon" />
    <h3 className="blank-slate-pf-main-action">{title}</h3>
    <p className="blank-slate-pf-secondary-action">{message}</p>
  </div>
);

WizardLoadingState.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string
};

WizardLoadingState.defaultProps = {
  message: ''
};

export default WizardLoadingState;
