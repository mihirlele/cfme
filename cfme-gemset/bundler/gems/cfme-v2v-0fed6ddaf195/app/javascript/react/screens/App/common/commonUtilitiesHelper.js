const commonUtilitiesHelper = {
  getMostRecentEntityByCreationDate: entities =>
    entities && entities.reduce((prev, current) => (prev.created_on > current.created_on ? prev : current)),

  groupBy: (items, key) =>
    items &&
    items.reduce(
      (result, item) => ({
        ...result,
        [item[key]]: [...(result[item[key]] || []), item]
      }),
      {}
    ),

  flatten: (arrayItems, flattenedArray) =>
    arrayItems && arrayItems.map(task => task.map(arrayElement => flattenedArray.push(arrayElement)))
};

export default commonUtilitiesHelper;
