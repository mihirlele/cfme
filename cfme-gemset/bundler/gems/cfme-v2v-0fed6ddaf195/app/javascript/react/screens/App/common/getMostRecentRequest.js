const getMostRecentRequest = requests =>
  requests.length && requests.reduce((prev, current) => (prev.created_on > current.created_on ? prev : current));

export default getMostRecentRequest;
