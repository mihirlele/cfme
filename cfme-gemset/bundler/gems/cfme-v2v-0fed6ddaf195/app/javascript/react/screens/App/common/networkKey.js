/**
 * returns a unique network based on current UI model
 */
export const networkKey = network => `${network.clusterId}\\${network.uid_ems || network.ems_ref}`;
