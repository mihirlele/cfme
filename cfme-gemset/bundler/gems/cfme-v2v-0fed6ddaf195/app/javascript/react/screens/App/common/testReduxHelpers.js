import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import configureMockStore from 'redux-mock-store';

const middlewares = [thunk, promiseMiddleware()];

export const generateStore = (reducers, initialState) =>
  createStore(combineReducers(reducers), initialState, applyMiddleware(...middlewares));

export const mockStore = initialState => configureMockStore(middlewares)(initialState);
