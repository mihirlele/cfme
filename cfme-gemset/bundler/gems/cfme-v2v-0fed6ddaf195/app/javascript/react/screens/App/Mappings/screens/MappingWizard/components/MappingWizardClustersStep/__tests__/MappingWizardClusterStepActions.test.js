import { mockStore } from '../../../../../../common/testReduxHelpers';
import * as actions from '../MappingWizardClustersStepActions';
import { initialState, requestSourceClustersData } from '../mappingWizardClustersStep.fixtures';
import { mockRequest, mockReset } from '../../../../../../../../../common/mockRequests';

const store = mockStore(initialState);

afterEach(() => {
  store.clearActions();
  mockReset();
});

describe('mappingWizard actions', () => {
  it('should fetch source clusters and return PENDING and FULFILLED action', () => {
    const { fetchSourceClustersUrl } = requestSourceClustersData;
    mockRequest({
      url: fetchSourceClustersUrl,
      status: 200
    });
    return store.dispatch(actions.fetchSourceClustersAction(fetchSourceClustersUrl)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  it('should fetch source clusters and return PENDING and REJECTED action', () => {
    const { fetchSourceClustersUrl } = requestSourceClustersData;
    mockRequest({
      url: fetchSourceClustersUrl,
      status: 404
    });
    return store.dispatch(actions.fetchSourceClustersAction(fetchSourceClustersUrl)).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});

describe('queryProvidersAction', () => {
  const url = '/api';
  const providerIds = ['1'];
  test('dispatches PENDING and FULFILLED actions', () => {
    mockRequest({
      method: 'POST',
      url,
      status: 200,
      response: [{ mock: 'data' }]
    });

    return store.dispatch(actions.queryProvidersAction(url, providerIds)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  test('dispatches PENDING and REJECTED actions', () => {
    mockRequest({
      method: 'POST',
      url,
      status: 404,
      response: { error: 'error' }
    });

    return store.dispatch(actions.queryProvidersAction(url, providerIds)).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
