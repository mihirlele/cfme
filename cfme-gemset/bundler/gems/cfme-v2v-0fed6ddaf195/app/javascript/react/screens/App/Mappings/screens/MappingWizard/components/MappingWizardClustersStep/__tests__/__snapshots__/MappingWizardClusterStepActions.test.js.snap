// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mappingWizard actions should fetch source clusters and return PENDING and FULFILLED action 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_SOURCE_CLUSTERS_PENDING",
  },
  Object {
    "type": "FETCH_V2V_SOURCE_CLUSTERS_FULFILLED",
  },
]
`;

exports[`mappingWizard actions should fetch source clusters and return PENDING and REJECTED action 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_SOURCE_CLUSTERS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "FETCH_V2V_SOURCE_CLUSTERS_REJECTED",
  },
]
`;

exports[`queryProvidersAction dispatches PENDING and FULFILLED actions 1`] = `
Array [
  Object {
    "type": "QUERY_V2V_PROVIDERS_PENDING",
  },
  Object {
    "payload": Array [
      Object {
        "mock": "data",
      },
    ],
    "type": "QUERY_V2V_PROVIDERS_FULFILLED",
  },
]
`;

exports[`queryProvidersAction dispatches PENDING and REJECTED actions 1`] = `
Array [
  Object {
    "type": "QUERY_V2V_PROVIDERS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "QUERY_V2V_PROVIDERS_REJECTED",
  },
]
`;
